src/ppx/geometry.cpp

```diff
@@ -19,6 +19,334 @@
+    }
+};
+
+class VertexDataProcessorInterleaved : public IVertexDataProcessor
+{
+public:
+    virtual bool Validate(Geometry* pGeom) override
+    {
+        return true;
+    }
+    virtual Result UpdateVertexBuffer(Geometry* pGeom) override
+    {
+        PPX_ASSERT_MSG((1 == GetVertexBindingCount(pGeom)), "there should be only 1 binding for planar");
+        AddVertexBuffer(pGeom, kBufferIndex);
+        return ppx::SUCCESS;
+    }
+    virtual uint32_t AppendVertexData(Geometry* pGeom, const TriMeshVertexData& vtx) override
+    {
+        uint32_t       startSize = GetVertexBufferSize(pGeom, kBufferIndex);
+        const uint32_t attrCount = GetVertexBindingAttributeCount(pGeom, kBufferIndex);
+        for (uint32_t attrIndex = 0; attrIndex < attrCount; ++attrIndex) {
+            const grfx::VertexSemantic semantic = GetVertexBindingAttributeSematic(pGeom, kBufferIndex, attrIndex);
+
+            // clang-format off
+        switch (semantic) {
+            default: break;
+            case grfx::VERTEX_SEMANTIC_POSITION  : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.position); break;
+            case grfx::VERTEX_SEMANTIC_NORMAL    : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.normal); break;
+            case grfx::VERTEX_SEMANTIC_COLOR     : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.color); break;
+            case grfx::VERTEX_SEMANTIC_TANGENT   : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.tangent); break;
+            case grfx::VERTEX_SEMANTIC_BITANGENT : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.bitangent); break;
+            case grfx::VERTEX_SEMANTIC_TEXCOORD  : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.texCoord); break;
+        }
+            // clang-format on
+        }
+        uint32_t endSize = GetVertexBufferSize(pGeom, kBufferIndex);
+
+        uint32_t bytesWritten = (endSize - startSize);
+        PPX_ASSERT_MSG((bytesWritten == GetVertexBufferElementSize(pGeom, kBufferIndex)), "size of vertex data written does not match buffer's element size");
+
+        return GetVertexBufferElementCount(pGeom, kBufferIndex);
+    }
+    virtual uint32_t AppendVertexData(Geometry* pGeom, const WireMeshVertexData& vtx) override
+    {
+        uint32_t       startSize = GetVertexBufferSize(pGeom, kBufferIndex);
+        const uint32_t attrCount = GetVertexBindingAttributeCount(pGeom, kBufferIndex);
+        for (uint32_t attrIndex = 0; attrIndex < attrCount; ++attrIndex) {
+            const grfx::VertexSemantic semantic = GetVertexBindingAttributeSematic(pGeom, kBufferIndex, attrIndex);
+
+            // clang-format off
+        switch (semantic) {
+            default: break;
+            case grfx::VERTEX_SEMANTIC_POSITION  : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.position); break;
+            case grfx::VERTEX_SEMANTIC_COLOR     : AppendDataToVertexBuffer(pGeom, kBufferIndex, vtx.color); break;
```

Comment:
Needs indention to match clang-format tags.

Reply:
fixed